### PR TITLE
V rmanda/183969/sample sdk/app only writes sdk sample02

### PR DESCRIPTION
**Pull request by v-rmanda**
Created "AssignmentResource" Method and created a Draft Assignment
- Created AP Setup Assignment resources folder 
- Posted the resources - AO add word resources to assignment
- Deleted the newly created assignment

### CODE REVIEW

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs [hunk 1]
```diff
@@ -14,5 +14,7 @@
 
 // Create an instance of the class you want to test and call the desired workflow method.
 // Each flow represents a common entire process that can be tested using Microsoft Graph SDK, all the sample flows are located in the "workflows" folder.
-ModuleWorkflow workflow = new ModuleWorkflow(config);
-await workflow.ClassworkAsync();
\ No newline at end of file
+
+AssignmentWorkflow workflow = new AssignmentWorkflow(config);
+// await workflow.ClassworkAsync(appOnly:true);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Program.cs]: remove commented line

File: README.md
Comment (cristobal-buenrostro) [inline README.md:1]: REVERT the changes in this file, you already updated the file in this PR.
https://github.com/microsoft/edu-assignments-graph-sdk/pull/16/files

you will have to update the README after you get all the samples merged

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 2]
```diff
@@ -34,7 +34,7 @@ public async Task<IEnumerable<EducationAssignment>> GetMeAssignmentsFromNonArchi
                 string userAccount = isTeacher ? _config["teacherAccount"] : _config["studentAccount"];
 
                 // Get a Graph client using delegated permissions
-                var graphClient = GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], userAccount, _config["password"]);
+                var graphClient = MicrosoftGraphSDK.GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], userAccount, _config["password"]);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: REVERT, already covered in https://github.com/microsoft/edu-assignments-graph-sdk/pull/16/files

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 3]
```diff
@@ -123,5 +123,87 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+
+                //Deleting the created assignment
+                await Assignment.DeleteAsync(graphClient, _config["classId"], assignmentId);
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"CreateAndPatchAssignment: {ex.ToString()}");
+            }
+        }
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: REVERT you already added this method in PR https://github.com/microsoft/edu-assignments-graph-sdk/pull/16/files.
you are mixing changes

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 4]
```diff
@@ -123,5 +123,87 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+
+        public async Task AssignmentResource(bool appOnly = false)
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: add a proper header with description and parameters

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 5]
```diff
@@ -123,5 +123,87 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+        {
+            try
+            {
+                string assignmentId = string.Empty;
+
+                // Get a Graph client using delegated permissions
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: // Get a Graph client based on the appOnly parameter

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 6]
```diff
@@ -123,5 +123,87 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                var graphClient = appOnly ? GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]) : GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                // Create assignment
+                var assignment = await Assignment.CreateAsync(graphClient, _config["classId"]);
+                assignmentId = assignment.Id;
+                Console.WriteLine($"Assignment created successfully {assignment.Id} in state {assignment.Status}");
+
+                //AO Set Up Assignment Resources Folder
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: // Set Up Assignment Resources Folder

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 7]
```diff
@@ -123,5 +123,87 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                await Assignment.SetupResourcesFolder(graphClient, _config["classId"], assignmentId);
+                Console.WriteLine("SetupResourceFolder creation successful");
+
+                //Adding AO Word Resource to assignment with distribute to student
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: // Adding new Word resource to assignment

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 4 above)
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: AssignmentResourceAsync

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 8]
```diff
@@ -123,5 +123,87 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                Console.WriteLine($"Resource created successfully {resource.Id} Display Name {resource.Resource.DisplayName}");
+
+                //Deleting the created assignment
+                await Assignment.DeleteAsync(graphClient, _config["classId"], assignmentId);
+                Console.WriteLine("Assignment deleted successfully");
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"AssignmentResource: {ex.ToString()}");
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: AssignmentResourceAsync

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:1]: REVERT all changes from this file, you already did it in PR https://github.com/microsoft/edu-assignments-graph-sdk/pull/16/files.
you are mixing changes

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 9]
```diff
@@ -239,5 +239,48 @@ public class Assignment
...
+        {
+            try
+            {               
+                await client.Education.Classes[classId].Assignments[assignmentId].DeleteAsync();          
+            }
+            catch (Exception ex)
+            {
+                throw new GraphException($"DeleteAsync call: {ex.Message}", ex, classId, assignmentId);
+            }
+        }
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]: REVERT the **PatchAsync** and **DeleteAsync**. You already added these methods in PR https://github.com/microsoft/edu-assignments-graph-sdk/pull/16/files.
you are mixing changes

**Review by cristobal-buenrostro — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs (hunk 1 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Program.cs]: removed the commented line "// await workflow.ClassworkAsync(appOnly:true);"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 4 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Updated to "/// <summary>
/// Workflow to create Assignment resources in resource folder
/// </summary>
/// <param name="appOnly">True value authenticates the graph client with application permissions only, otherwise it will be created with delegated permissions.</param> "

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 5 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated to "// Get a Graph client based on the appOnly parameter"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 6 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated to "// Set Up Assignment Resources Folder"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 7 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated to "// Adding new Word resource to assignment"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 8 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated to "AssignmentResourceAsync"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 4 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated the method name to "AssignmentResourceAsync"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 2 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: reverted the changes

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 3 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: reverted the changes

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 9 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs]: removed PatchAsync and DeleteAsync methods

File: README.md
Comment (cristobal-buenrostro) [inline README.md:1]: again, revert all changes in this file.

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:1]: again, revert all changes in this file

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 2 above)
Comment (cristobal-buenrostro) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: you didn't revert it, apply the suggestion

```suggestion
                var graphClient = GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], userAccount, _config["password"]);
```

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 10]
```diff
@@ -123,5 +123,91 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                //Deleting the created assignment
+                await Assignment.DeleteAsync(graphClient, _config["classId"], assignmentId);
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"CreateAndPatchAssignment: {ex.ToString()}");
+            }
+        }
+
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: just remove it, apply the suggestion

```suggestion

```

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 4 above)
Comment (cristobal-buenrostro) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: one more time

```suggestion
        public async Task AssignmentResourceAsync(bool appOnly = false)
```

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 9 above)
Comment (cristobal-buenrostro) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs]: please double check I still see them, no changes needed in this file

**Comment by cristobal-buenrostro** [comment]
@v-rmanda 
you resolved some of the comments but the changes were not reflected please double check

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:1]: reverted the changes

Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:1]: reverted the changes

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 9 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs]: removed PatchAsync method

File: README.md
Comment (v-rmanda) [reply README.md:1]: reverted the changes on Readme.md file

Comment (v-rmanda) [reply README.md:1]: reverted the changes of Readme.md file.

File: README.md [hunk 11]
```diff
@@ -119,6 +119,8 @@ Use the `Program.cs` file to test any of the current workflows.
 |4| Assignment evolvable enums   | Evolvable enums is a mechanism that Microsoft Graph API uses to add new members to existing enumerations without causing a breaking change for applications.|[View](https://github.com/microsoft/edu-assignments-graph-sdk/blob/main/samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs#L68)|
 |5| Classwork Module   | The sample shows how the teacher creates a module and attach resources and assignments. |[View](https://github.com/microsoft/edu-assignments-graph-sdk/blob/main/samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs#L26)|
 
+
+
```
Comment (cristobal-buenrostro) [inline README.md]: nit: remove these two empties lines.

**Review by cristobal-buenrostro — APPROVED** [review]
Approved, just take care of the open comment

File: README.md (hunk 11 above)
Comment (v-rmanda) [reply README.md]: removed empty lines

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 12]
```diff
@@ -123,5 +123,53 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
             }
         }
-    }
+
+        /// <summary>
+        /// Workflow to create Assignment resources in resource folder
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Workflow to create Assignment resources under the assignment resource folder

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 13]
```diff
@@ -123,5 +123,53 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+        {
+            try
+            {
+                string assignmentId = string.Empty;
+
+                // Get a Graph client based on the appOnly parameter
+                var graphClient = appOnly ? GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]) : GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                // Create assignment
+                var assignment = await Assignment.CreateAsync(graphClient, _config["classId"]);
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Call it CreateSampleAssignmentAsync

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 14]
```diff
@@ -123,5 +123,53 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                assignmentId = assignment.Id;
+                Console.WriteLine($"Assignment created successfully {assignment.Id} in state {assignment.Status}");
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs:143]: Assignment created successfully with Id: {assignment.Id} in state: {assignment.Status}"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 15]
```diff
@@ -123,5 +123,53 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                    Resource = new EducationWordResource
+                    {
+                        OdataType = "microsoft.graph.educationWordResource",
+                        DisplayName = "New Word Document.docx",                        
+                    },
+                };
+
+                var resource = await Assignment.PostResourceAsync(graphClient, _config["classId"], assignmentId, requestBody);
+
+                Console.WriteLine($"Resource created successfully {resource.Id} Display Name {resource.Resource.DisplayName}");
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: $"Resource created successfully with id : {resource.Id},  Display Name : {resource.Resource.DisplayName}"

**Review by sharad-sharma-msft — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 12 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated to "Workflow to create Assignment resources under the assignment resource folder"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 14 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs:143]: updated to "Assignment created successfully with Id: {assignment.Id} in state: {assignment.Status}"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 15 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated to "$"Resource created successfully with id : {resource.Id}, Display Name : {resource.Resource.DisplayName}"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 13 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: replaced createAsync method with "CreateSampleAssignmentAsync" for 15 occurences in solution.

File: samples/csharp/MicrosoftEduGraphSamples/MicrosoftEduGraphSamples.csproj [hunk 16]
```diff
@@ -12,6 +12,7 @@
     <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="6.0.0" />
     <PackageReference Include="Microsoft.Graph" Version="5.21.0" />
     <PackageReference Include="Microsoft.Graph.Beta" Version="5.46.0-preview" />
+    <PackageReference Include="Neo4jClient" Version="5.1.14" />
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/MicrosoftEduGraphSamples.csproj]: where is this dependency coming from?

Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/MicrosoftEduGraphSamples.csproj]: That was added while trying to fix an issue. Removed as its no more needed.

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 17]
```diff
@@ -123,5 +123,53 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
             }
         }
-    }
+
+        /// <summary>
+        /// Workflow to create Assignment resources under the assignment resource folder
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: similar comment (Please make this change across all code samples)

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 18]
```diff
@@ -123,5 +123,53 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                // Get a Graph client based on the appOnly parameter
+                var graphClient = appOnly ? GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]) : GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                // Create assignment
+                var assignment = await Assignment.CreateSampleAssignmentAsync(graphClient, _config["classId"]);
+                assignmentId = assignment.Id;
+                Console.WriteLine($"Assignment created successfully with Id: {assignment.Id} in state: {assignment.Status}");
+
+                // Set Up Assignment Resources Folder
+                await Assignment.SetupResourcesFolder(graphClient, _config["classId"], assignmentId);
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: SetupResourcesFolderAsync

**Review by sharad-sharma-msft — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 17 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated workflow to "A code sample"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 18 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Changed "SetupResourcesFolder" to "SetupResourcesFolderAsync"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 19]
```diff
@@ -9,7 +9,7 @@
 namespace MicrosoftEduGraphSamples.Workflows
 {
     /// <summary>
-    /// Contains all the workflows related to Assignments, include getting assignments from all classes, 
+    /// Contains all the code samples related to Assignments, include getting assignments from all classes,
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: nit: avoid saying "all"

**Review by sharad-sharma-msft — APPROVED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 19 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: removed all and updated to " /// Contains the code samples related to Assignments, include getting assignments from all classes, "

**Review by sharad-sharma-msft — APPROVED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 20]
```diff
@@ -80,16 +80,20 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 var graphClient = GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
 
                 // Create assignment to verify inactive state
-                var assignmentInactive = await Assignment.CreateSampleAsync(graphClient, _config["classId"]);
+
+                var assignmentInactive = await Assignment.CreateSampleAssignmentAsync(graphClient, _config["classId"]);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs:84]: nit: remove the blank line

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 21]
```diff
@@ -80,16 +80,20 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+
                 assignmentId = assignmentInactive.Id;
                 Console.WriteLine($"Assignment created successfully {assignmentInactive.Id} in state {assignmentInactive.Status}");
 
                 // Create assignment to verify assigned state
-                var assignmentAssigned = await Assignment.CreateSampleAsync(graphClient, _config["classId"]);
+
+                var assignmentAssigned = await Assignment.CreateSampleAssignmentAsync(graphClient, _config["classId"]);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs:91]: nit: remove the blank line

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 22]
```diff
@@ -125,10 +129,12 @@ public async Task AssignmentsEvolvableEnumsAsync()
         }
 
         /// <summary>
-        /// A code sample to create a new assignment, followed by an update to the assignment display name (aka title)
+
+        /// A code sample to create Assignment resources under the assignment resource folder
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs:133]: nit: remove blank line

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 23]
```diff
@@ -125,10 +129,12 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
         /// </summary>
         /// <param name="appOnly">True value authenticates the graph client with application permissions only, otherwise it will be created with delegated permissions.</param> 
-        public async Task CreateAndPatchAssignmentAsync(bool appOnly = false)
+        public async Task AssignmentResourceAsync(bool appOnly = false)
+
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs:137]: nit: remove blank line

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 24]
```diff
@@ -138,31 +144,39 @@ public async Task CreateAndPatchAssignmentAsync(bool appOnly = false)
                 var graphClient = appOnly ? GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]) : GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
 
                 // Create assignment
-                var assignment = await Assignment.CreateSampleAsync(graphClient, _config["classId"]);
+
+                var assignment = await Assignment.CreateSampleAssignmentAsync(graphClient, _config["classId"]);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs:148]: nit: remove blank line

**Review by cristobal-buenrostro — COMMENTED** [review]
please revert the last two commits and resolve the conflicts carefully.

if you revert the commits, then not need to take care of the open comments to remove the blank lines manually.

**Review by cristobal-buenrostro — CHANGES_REQUESTED** [review]
please follow my suggestions above

**Review by cristobal-buenrostro — APPROVED** [review]
